app/services/queue_listener.rb

```diff
@@ -101,4 +101,26 @@ def update_member(params)
       raise Error.new("Member update failed with #{res.parsed_response["errors"]}!")
     end
   end
+
+  def action_params(params)
+    country = params[:params][:country]
+    raise Error.new("Generic action processing attempted without a country in the request.") unless country.present?
+    if country == "United States" && ENV['BYPASS_ZIP_VALIDATION']
+      raise Error.new("Your default US zip code has not been set!") unless ENV['DEFAULT_US_ZIP'].present?
+      params[:params][:postal] = verify_zip(params[:params][:postal])
+    end
+    params
+  end
```

Comment:
I kinda agree, and all of this could essentially be moved to ActionCreator, which could be used for `subscribe_member`, as well, but that would also take a pretty non-trivial amount of refactoring, since the `run` method does too many things. So either I refactor them here, or I refactor them there, while considering how much of this is really worth doing since it doesn't really progress product priorities and is just an AK API wrapper service.